Tests/bip32-ed25519-swiftTests/bip32-ed25519-swiftTests.swift

```diff
@@ -125,25 +125,121 @@ final class Bip32Ed25519Tests: XCTestCase {
+        let outputPublic = c!.deriveKey(rootKey: rootkey, bip44Path: bip44Path, isPrivate: false, derivationType: BIP32DerivationType.Khovratovich)
+        XCTAssertEqual(outputPublic.prefix(32), expectedResultPublic)
 
         let expectedResultPrivate = Data([128, 16, 43, 185, 143, 170, 195, 253, 23, 137, 194, 198, 197, 89, 211, 113, 92, 217, 202, 194, 40, 214, 212, 176, 247, 106, 35, 70, 234, 236, 143, 70, 1, 174, 20, 40, 64, 137, 36, 62, 147, 107, 233, 27, 40, 35, 204, 20, 47, 117, 49, 53, 234, 255, 27, 174, 32, 211, 238, 199, 120, 112, 197, 68, 159, 146, 199, 144, 215, 171, 174, 224, 224, 10, 78, 193, 251, 120, 161, 212, 56, 232, 204, 247, 194, 186, 217, 160, 24, 165, 191, 154, 93, 81, 0, 117])
-        let outputPrivate = c!.deriveKey(rootKey: rootkey, bip44Path: bip44Path, isPrivate: true)
+        let outputPrivate = c!.deriveKey(rootKey: rootkey, bip44Path: bip44Path, isPrivate: true, derivationType: BIP32DerivationType.Khovratovich)
         XCTAssertEqual(outputPrivate, expectedResultPrivate)
     }
 
+    func testDeriveChildNodePublic() throws {
+        let seed = try Mnemonic.deterministicSeedString(from: "salon zoo engage submit smile frost later decide wing sight chaos renew lizard rely canal coral scene hobby scare step bus leaf tobacco slice")
+        guard let data = Data(hexString: seed) else {
+            return
+        }
+
+        let context = KeyContext.Address
+        let account: UInt32 = 0
+        let change: UInt32 = 0
+
+        let bip44Path: [UInt32] = [c!.harden(44), c!.harden(283), c!.harden(0), 0]
+
+        let walletRoot = c!.deriveKey(
+            rootKey: c!.fromSeed(data),
+            bip44Path: bip44Path,
+            isPrivate: false,
+            derivationType: BIP32DerivationType.Peikert
+        )
+
+        // should be able to derive all public keys from this root without knowing private information
+        // since these are SOFTLY derived
+
+        let numPublicKeysToDerive = 10
+        for i in 0 ..< numPublicKeysToDerive {
+            // assuming in a third party that only has public information
+            // I'm provided with the wallet level m'/44'/283'/0'/0 root [public, chaincode]
+            // no private information is shared
+            // i can SOFTLY derive N public keys / addresses from this root
+            let derivedKey = try c!.deriveChildNodePublic(extendedKey: walletRoot, keyIndex: UInt32(i), g: BIP32DerivationType.Peikert.rawValue)
+            // Deriving from my own wallet where i DO have private information
+            let myKey = c!.keyGen(context: context, account: account, change: change, keyIndex: UInt32(i), derivationType: BIP32DerivationType.Peikert)
+
+            // they should match
+            // derivedKey.subarray(0, 32) ==  public key (excluding chaincode)
+            XCTAssertEqual(derivedKey.prefix(32), myKey)
+        }
+    }
+
+    func testDeriveChildNodePublicHardenedChange() throws {
```

Comment:
In practice we would never harden the change

Reply:
Yup! for the most part just trying to ensure we have the same tests across languages and this was one Bruno had for the TS implementation

Reply:
Yeah we'll align all the tests, namings etc at the end when we move TS implementation into its own repo.